Source/utils/file_util.cpp

```diff
@@ -73,6 +73,27 @@ bool FileExists(const char *path)
 #endif
 }
 
+bool IsWriteable(const char *path)
+#if defined(_WIN64) || defined(_WIN32)
+{
+	const auto pathUtf16 = ToWideChar(path);
+	if (pathUtf16 == nullptr) {
+		LogError("UTF-8 -> UTF-16 conversion error code {}", ::GetLastError());
+		return false;
+	}
+	uint32_t attr = ::GetFileAttributesW(&pathUtf16[0]);
+	if (attr != INVALID_FILE_ATTRIBUTES && attr & FILE_ATTRIBUTE_READONLY) {
+		return false;
+	}
+	return true;
+}
+#else
+{
+	struct stat stats;
+	return stat(path, &stats) == 0 && (stats.st_mode & 2) != 0;

```

Comment:
1. What is this `2`? Should it be something like this instead? The values of `st_mode` are platform-specific.

  ```suggestion
	  return stat(path, &stats) == 0 && (stats.st_mode & S_IWUSR) != 0;
  ```

2. On platforms that support `access`, we should use it instead:

   ```c++
   #elif _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__)
	   return ::access(path, W_OK) == 0;
   #else
   ```